drivers/user_directory_report/app/controllers/user_directory_report/warehouse_reports/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module UserDirectoryReport::WarehouseReports
+  class UsersController < ApplicationController
+    helper_method :nav_link_classes
+    helper_method :cas_available?
+
+    def readonly?
+      true
+    end
+
+    def warehouse
+      @users = _users(User)
+      @user_source = 'warehouse'
+      render :report
+    end
+
+    def cas
+      if cas_available?
+        @users = _users(Cas::User)
+      else
+        @users = {}
```

Comment:
This should return `[]` if it fails I think.  Scopes essentially amount to arrays.

Reply:
✅